src/app/shipping/MultiShippingForm.tsx

```diff
@@ -87,32 +107,101 @@ class MultiShippingForm extends PureComponent<MultiShippingFormProps & WithLangu
+                    shouldShowErrorCode={ false }
+                /> }
+                { <AddressFormModal
+                    countries={ countries }
+                    countriesWithAutocomplete={ countriesWithAutocomplete }
+                    defaultCountryCode={ defaultCountryCode }
+                    getFields={ getFields }
+                    googleMapsApiKey={ googleMapsApiKey }
                     isLoading={ isLoading }
-                    isMultiShippingMode={ true }
-                    shouldDisableSubmit={ this.shouldDisableSubmit() }
-                    shouldShowOrderComments={ shouldShowOrderComments }
-                    shouldShowShippingOptions={ !hasUnassignedLineItems(consignments, cart.lineItems) }
-                />
-            </Form>
+                    isOpen={ !!itemAddingAddress }
+                    onRequestClose={ this.handleCloseAddAddressForm }
+                    onSaveAddress={ this.handleSaveAddress }
+                /> }
+
+                <Form>
+                    <ul className="consignmentList">
+                        { items.map(item => (
+                            <li key={ item.key }>
+                                <ItemAddressSelect
+                                    addresses={ addresses }
+                                    item={ item }
+                                    onSelectAddress={ this.handleSelectAddress }
+                                    onUseNewAddress={ this.handleUseNewAddress }
+                                />
+                            </li>
+                        )) }
+                    </ul>
+
+                    <ShippingFormFooter
+                        cartHasChanged={ cartHasChanged }
+                        isLoading={ isLoading }
+                        isMultiShippingMode={ true }
+                        shouldDisableSubmit={ this.shouldDisableSubmit() }
+                        shouldShowOrderComments={ shouldShowOrderComments }
+                        shouldShowShippingOptions={ !hasUnassignedLineItems(consignments, cart.lineItems) }
+                    />
+                </Form>
+            </Fragment>
         );
     }
 
+    private handleCloseErrorModal: () => void = () => {
+        this.setState({ createCustomerAddressError: undefined });
+    };
+
+    private handleSaveAddress: (address: AddressFormValues) => void = async address => {
+        const { createCustomerAddress, onUnhandledError } = this.props;
+        const { itemAddingAddress } = this.state;
+
+        if (!itemAddingAddress) {
+            return;
+        }
+
+        const shippingAddress = mapAddressFromFormValues(address);
+
+        await this.handleSelectAddress(shippingAddress, itemAddingAddress.itemId, itemAddingAddress.key);
+
+        try {
+            await createCustomerAddress(shippingAddress);
+        } catch (e) {
+            onUnhandledError(new CreateCustomerAddressError(e));
```

Comment:
To answer your concern regarding displaying the error as a modal. What about if we display the error as an `Alert` inside the modal, considering the error already happens inside a modal. The UX pattern is similar to manage payment instruments modal. And the user doesn't need to reopen the modal and retype everything to retry.

Another question that I have is, what happens if you refresh the page after receiving such error. I assume the dropdown won't be displaying an address because it's designed to display a list of "saved" addresses? So what would the experience be like?

Finally the error message doesn't adequately communicate that the shopper can continue to check out despite the fact that their address is not saved to their address book. Imagine you're a shopper who doesn't know the implementation, when I see that error message, would I still think that I can proceed? Or would I feel something's broken.